File: stack/Stack.java
class Stack {
    private static final int MAX = 5;  // Maximum size of the stack
    private int top;
    private int[] items;

    // Constructor to initialize the stack
    public Stack() {
        top = -1;
        items = new int[MAX];
    }

    // Function to check if the stack is full
    public boolean isFull() {
        return top == MAX - 1;
    }

    // Function to check if the stack is empty
    public boolean isEmpty() {
        return top == -1;
    }

    // Function to add an element to the stack (push operation)
    public void push(int value) {
        if (isFull()) {
            System.out.println("Stack Overflow! Cannot push " + value + " onto the stack.");
            return;
        }
        items[++top] = value;
        System.out.println(value + " pushed onto the stack.");
    }

    // Function to remove an element from the stack (pop operation)
    public int pop() {
        if (isEmpty()) {
            System.out.println("Stack Underflow! Cannot pop from the stack.");
            return -1;
        }
        int poppedItem = items[top--];
        System.out.println(poppedItem + " popped from the stack.");
        return poppedItem;
    }

    // Function to return the top element of the stack (peek operation)
    public int peek() {
        if (isEmpty()) {
            System.out.println("Stack is empty! Nothing to peek.");
            return -1;
        }
        return items[top];
    }

    // Function to display the stack
    public void display() {
        if (isEmpty()) {
            System.out.println("Stack is empty!");
            return;
        }
        System.out.println("Stack elements are: ");
        for (int i = top; i >= 0; i--) {
            System.out.println(items[i]);
        }
    }

    public static void main(String[] args) {
        Stack stack = new Stack();

        stack.push(10);
        stack.push(20);
        stack.push(30);
        stack.push(40);
        stack.push(50);
        stack.push(60);  // This will cause stack overflow

        stack.display();

        System.out.println("Top element is " + stack.peek());

        stack.pop();
        stack.pop();
        stack.pop();

        stack.display();

        System.out.println("Top element is " + stack.peek());
    }
}

File: stack/Stack.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 5 // Maximum size of the stack

typedef struct Stack
{
    int items[MAX];
    int top;
} Stack;

// Function to initialize the stack
void initialize(Stack *stack)
{
    stack->top = -1;
}

// Function to check if the stack is full
int isFull(Stack *stack)
{
    return stack->top == MAX - 1;
}

// Function to check if the stack is empty
int isEmpty(Stack *stack)
{
    return stack->top == -1;
}

// Function to add an element to the stack (push operation)
void push(Stack *stack, int value)
{
    if (isFull(stack))
    {
        printf("Stack Overflow! Cannot push %d onto the stack.\n", value);
        return;
    }
    stack->items[++(stack->top)] = value;
    printf("%d pushed onto the stack.\n", value);
}

// Function to remove an element from the stack (pop operation)
int pop(Stack *stack)
{
    if (isEmpty(stack))
    {
        printf("Stack Underflow! Cannot pop from the stack.\n");
        return -1;
    }
    int poppedItem = stack->items[(stack->top)--];
    printf("%d popped from the stack.\n", poppedItem);
    return poppedItem;
}

// Function to return the top element of the stack (peek operation)
int peek(Stack *stack)
{
    if (isEmpty(stack))
    {
        printf("Stack is empty! Nothing to peek.\n");
        return -1;
    }
    return stack->items[stack->top];
}

// Function to display the stack
void display(Stack *stack)
{
    if (isEmpty(stack))
    {
        printf("Stack is empty!\n");
        return;
    }
    printf("Stack elements are: \n");
    for (int i = stack->top; i >= 0; i--)
    {
        printf("%d\n", stack->items[i]);
    }
}

int main()
{
    Stack stack;
    initialize(&stack);

    push(&stack, 10);
    push(&stack, 20);
    push(&stack, 30);
    push(&stack, 40);
    push(&stack, 50);
    push(&stack, 60); // This will cause stack overflow

    display(&stack);

    printf("Top element is %d\n", peek(&stack));

    pop(&stack);
    pop(&stack);
    pop(&stack);

    display(&stack);

    printf("Top element is %d\n", peek(&stack));

    return 0;
}
